packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -182,58 +240,11 @@ export type SelectProps<T> = {
     'size' | 'disabled' | 'placeholder' | 'onChange'
   >;
 
-/**
- * Select component to select value from options.
- * Generic type is used for value. It prevents you from mistakenly using other types in `options`, `value` and `onChange`.
- * @example
- * <caption>Basic select.</caption>
- * <Select<number> options={[{ value: 1, label: 'Option 1' }, { value: 2, label: 'Option 2' }, { value: 3, label: 'Option 3' }]} />
- * @example
- * <caption>Disabled select with placeholder.</caption>
- * <Select
- *   disabled={true}
- *   placeholder='Placeholder text'
- *   options={[{ value: 1, label: 'Option 1' }, { value: 2, label: 'Option 2' }, { value: 3, label: 'Option 3' }]}
- * />
- * @example
- * <caption>Select with selected value and change handler.</caption>
- * <Select<number>
- *   value={selectedValue}
- *   onChange={(value) => setSelectedValue(value)}
- *   options={[{ value: 1, label: 'Option 1' }, { value: 2, label: 'Option 2' }, { value: 3, label: 'Option 3' }]}
- * />
- * @example
- * <caption>Select using custom renderers for menu items and selected value.</caption>
- * <Select<string>
- *   options={[
- *     { value: 'yellow', label: 'Yellow' },
- *     { value: 'green', label: 'Green' },
- *     { value: 'red', label: 'Red' },
- *   ]}
- *   value={selectedValue}
- *   placeholder='Placeholder text'
- *   itemRenderer={(option, itemProps) => (
- *     <MenuItem
- *       style={{ color: option.value }}
- *       isSelected={itemProps.isSelected}
- *       onClick={() => {
- *         setSelectedValue(option.value);
- *         itemProps.close();
- *       }}
- *       role='option'
- *       ref={(el) => itemProps.isSelected && el?.scrollIntoView()}
- *     >
- *       {option.label}
- *     </MenuItem>
- *   )}
- *   selectedItemRenderer={(option) => (
- *     <span style={{ backgroundColor: option.value }}>{option.label}</span>
- *   )}
- * />
- */
-export const Select = React.forwardRef(
+// ----------------------------------------------------------------------------
+
+const CustomSelect = React.forwardRef(
```

Comment:
The order of the components and types in the file can be made more consistent to make it more easy to read. E.g. `Component` and then `ComponentProps` (or vice versa), helper components like `SelectButton`/`SelectEndIcon` at the bottom of the file, etc.

Reply:
Sure, I can reorder the thing towards the end, once most of the review is done. I'll leave it as-is for now to minimize diff.

Reply:
Sone [`4a2fc3d`](https://github.com/iTwin/iTwinUI/pull/1879/commits/4a2fc3d88d51a1631865c250e3c2da4a07b67b1f). The most important things are now at the top, with types for it immediately below, and so on